sdk/storage/Azure.Storage.Common/tests/Shared/StorageTestBase.cs

```diff
@@ -28,13 +29,69 @@ static StorageTestBase()
 #endif
         }
 
+        /// <summary>
+        /// Add a static TestEventListener which will redirect SDK logging
+        /// to Console.Out for easy debugging.
+        /// </summary>
+        private static TestEventListener s_listener;
+
         public StorageTestBase(bool async, RecordedTestMode? mode = null)
             : base(async, mode ?? RecordedTestUtilities.GetModeFromEnvironment())
         {
             Sanitizer = new StorageRecordedTestSanitizer();
             Matcher = new StorageRecordMatcher(Sanitizer);
         }
 
+        /// <summary>
+        /// Start logging events to the console if debugging or in Live mode.
+        /// This will run once before any tests.
+        /// </summary>
+        [OneTimeSetUp]
+        public void StartLoggingEvents()
+        {
+            if (Debugger.IsAttached || Mode == RecordedTestMode.Live)
+            {
+                s_listener = new TestEventListener();
+            }
+        }
+
+        /// <summary>
+        /// Stop logging events and do necessary cleanup.
+        /// This will run once after all tests have finished.
+        /// </summary>
+        [OneTimeTearDown]
+        public void StopLoggingEvents()
+        {
+            s_listener?.Dispose();
```

Comment:
Nit - null it out after disposing so your other checks for null won't ever make calls on a disposed instance.

Reply:
Fixed.